### ISSUE TITLE
[NodeBundle] link select language problem

### ISSUE THREAD

**Issue by slaci** [Kunstmaan/KunstmaanBundlesCMS#1159]
The url chooser works only with multilanguage sites now: https://github.com/Kunstmaan/KunstmaanBundlesCMS/blob/master/src/Kunstmaan/NodeBundle/Resources/views/Widgets/selectLink.html.twig#L83

In this file: `data-base-url={{ "/" ~ app.request.locale }}` is the problematic piece, cause locale is not needed in the url on single language settings.

The fix should be simple, just `{{ multilanguage ? "/" ~ app.request.locale : '' }}`, but that multilanguage parameter is not accessible in twig (and cant access the container either).

The other parameters are defined in [config.yml](https://github.com/Kunstmaan/KunstmaanBundlesStandardEdition/blob/master/app/config/config.yml#L17), which is in a different package, and putting the `multilanguage: %multilanguage%` line there would be very easy but also a bc break.

So, what do you think, is it okey to fix this way or have a better idea?

---
Other approach would be a prod url generator, which just replaces the scriptname from the generated url (this is big hax and not recommended). A twig extension in NodeTwigExtension would be for this (with needs_context => true):

```php
public function getProdPathForNodeUrl($context, $urlOrNode, $locale = null, $routeParams = array(), $relative = false, $prodEnv = 'prod')
    {
        $request = $this->requestStack->getMasterRequest();

        if ($urlOrNode instanceof Node) {
            $urlOrNode->getNodeTranslation($locale);
        }

        if ($urlOrNode instanceof NodeTranslation) {
            $locale = $locale ?: $urlOrNode->getLang();
            $urlOrNode = $urlOrNode->getUrl();
        }

        $routeParams = array_merge($routeParams, array(
            'url'       => $urlOrNode,
            '_locale'   => $locale ?: $request->attributes->get('_locale'),
        ));

        $path = $this->generator->generate(
            '_slug',
            $routeParams,
            $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH
        );

        if ($context['app']->getEnvironment() !== $prodEnv) {
            $path = str_replace($request->getScriptName(), '', $path);
        }

        return $path;
    }
```

**Comment by jockri** [comment]
I think we can even do this for getting the base url:

```
{{ path('_slug', {'url': ''}) }}
```

**Comment by slaci** [comment]
But when some1 select the link in dev mode, then the base will be /app_dev.php/ and that will be saved I guess. This is why i posted that ugly scriptname remove hack (but still don't like it).

**Comment by iknowfoobar** [comment]
I just submitted a pull request that fixes this #1224
